scilpy/tractanalysis/distance_to_centroid.py

```diff
@@ -1,38 +1,536 @@
+
+    indices = np.array(np.nonzero(binary_mask), dtype=int).T
+    labels = min_dist_to_centroid(indices,
+                                  sft_centroid[0].streamlines._data,
+                                  nb_pts=nb_pts)
+    logging.debug('Computed labels using the euclidian method '
+                  f'in {round(time.time() - timer, 3)} seconds')
+    min_label, max_label = labels.min(), labels.max()
+
+    if method == 'centerline':
+        labels_map = np.zeros(binary_mask.shape, dtype=np.uint16)
+        labels_map[tuple(indices.T)] = labels
+    elif method == 'hyperplane':
+        min_label, max_label = labels.min(), labels.max()
+        del labels, indices
+        logging.debug('Computing Labels using the hyperplane method.\n'
+                      '\tThis can take a while...')
+        # Select 2000 elements from the SFTs to train the classifier
+        streamlines_length = [length(streamline)
+                              for streamline in sft.streamlines]
+        random_indices = np.random.choice(len(sft.streamlines), 2000)
+        tmp_sft = resample_streamlines_step_size(
+            sft[random_indices], np.min(streamlines_length) / nb_pts)
+
+        # Associate the labels to the streamlines using the centroids as
+        # reference (to handle shorter bundles due to missing data)
+        mini_timer = time.time()
+        labels, points, = associate_labels(tmp_sft, min_label, max_label)
+
+        kd_tree = KDTree(points)
+        indices = np.array(np.nonzero(binary_mask), dtype=int).T
+
+        nn_indices = kd_tree.query(indices, k=1)[1]
+        labels, points = labels[nn_indices], points[nn_indices]
+
+        logging.debug('\tAssociated labels to centroids in '
+                      f'{round(time.time() - mini_timer, 3)} seconds')
+
+        # Initialize the scaler
+        mini_timer = time.time()
+        scaler = MinMaxScaler(feature_range=(-1, 1))
+        scaler.fit(points)
+        scaled_streamline_data = scaler.transform(points)
+
+        svc = SVC(C=1.0, kernel='rbf', random_state=1)
+
+        svc.fit(X=scaled_streamline_data, y=labels)
+        logging.debug('\tSVC fit of training data in '
+                      f'{round(time.time() - mini_timer, 3)} seconds')
+
+        # Scale the coordinates of the voxels
+        mini_timer = time.time()
+        masked_binary_mask = np.zeros(binary_mask.shape, dtype=np.uint8)
+        masked_binary_mask[::2, ::2, ::2] = binary_mask[::2, ::2, ::2]
+        voxel_coords = np.array(np.where(masked_binary_mask)).T
+        scaled_voxel_coords = scaler.transform(voxel_coords)
+
+        # Predict the labels for the voxels
+        labels = svc.predict(X=scaled_voxel_coords)
+        logging.debug('\tSVC prediction of labels in '
+                      f'{round(time.time() - mini_timer, 3)} seconds')
+
+        logging.debug('Computed labels using the hyperplane method '
+                      f'in {round(time.time() - timer, 3)} seconds')
```

Comment:
There is still some processing being done after. Perharps move this timer to line 519 ?